scripts/cosine_similarity_preprocessing.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python
+
+'''
+Script to parse through all yelp users and create a mapping
+from restaurant_id => [cosine similarity matrix, userid -> index into matrix dictionary] 
+Writes to 'restaurants_cosine_similarity.json' file as json
+'''
+
+import json
+import os
+from itertools import islice
+from itertools import combinations
+from collections import OrderedDict
+
+import numpy as np
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+
+from init_friend_graph import init_friend_graph
+
+import pickle
+
+restaurants = {}
+threshold = 0.5
```

Comment:
Yea I'll try that.  I would have thought that would lead to more null values not less, which is why I decided to start at .5, but definitely worth a try.